skyportal/handlers/api/candidate.py

```diff
@@ -1124,7 +1189,22 @@ def grab_query_results(
 
     items = []
     query_options = [joinedload(Obj.thumbnails)] if include_thumbnails else []
-    for item_id in page_ids:
-        items.append(Obj.query.options(query_options).get(item_id))
+
+    if len(page_ids) > 0:
+        page_ids_values = get_obj_id_values(page_ids)
+        items = (
+            DBSession()
+            .query(Obj)
+            .options(query_options)
+            .join(page_ids_values, page_ids_values.c.id == Obj.id)
+            .order_by(page_ids_values.c.ordering)
+        )
+    else:
+        # If there are no values, the VALUES statement above will cause a syntax error,
+        # so just switch to this simple IN query. Should return nothing either way
+        items = (
+            DBSession().query(Obj).options(query_options).filter(Obj.id.in_(page_ids))
+        )
```

Comment:
The change here is that instead of looping through the ordered Obj IDs to be returned and manually building up a Python list of the Objs, we return a Query object against `Obj`, with the order maintained by joining against a `VALUES` table. This accomplished two goals:
- All the Obj entities are fetched in one go instead of doing a bunch of `Obj.query.get()` queries.
- Returning a SQLAlchemy Query here instead of a standard Python list means I can actually do the `add_columns()` stuff with the last and peak detections I did in `sources.py` to speed that part up.